tests/components/media_player/test_init.py

```diff
@@ -339,6 +342,70 @@ async def test_media_browse(
     assert msg["result"] == {"bla": "yo"}
 
 
+async def test_media_browse_service(hass: HomeAssistant) -> None:
+    """Test browsing media using service call."""
+    await async_setup_component(
+        hass, "media_player", {"media_player": {"platform": "demo"}}
+    )
+    await hass.async_block_till_done()
+
+    with patch(
+        "homeassistant.components.media_player.MediaPlayerEntity.async_browse_media",
```

Comment:
I'd expect to patch the implementing entity, or implement this method in that entity, eg here it's the demo integration media player entity. Ie don't patch the base `MediaPlayerEntity`.

Reply:
Good point, I've changed it.  Since the demo did not implement async_browse_media it was using the default implementation in the MediaPlayerEntity. I copied the code from a prior test in this module (we should update that in a separate PR)

https://github.com/home-assistant/core/blob/5c0ea9e845c9b6d6c11246af328b8bc4fd1df04d/tests/components/media_player/test_init.py#L322-L325